src/service/google-sheet.service.ts

```diff
@@ -0,0 +1,331 @@
+import { GoogleSpreadsheetsService } from './google-spreadsheet.service';
+import {
+    SpreadsheetDataModel,
+    SpreadsheetDataValueModel
+} from './google-spreadsheet.service';
+interface SheetConfig {
+    spreadsheetId: string;
+    sheetName: string;
+    headersList?: string[];
+}
+
+export class GoogleSheetService<T extends SpreadsheetDataModel> {
```

Comment:
I think at least giving the user a way to input a range is valuable, since the way you currently have the sheet set up there at sometime multiple sheets with in one. For example the core sheet has meta data, meetings & roles.

Reply:
The constructor takes a sheet name argument which means that each instance is one sheet. The way it was I think requires that each instance be limited to a range within a sheet which is too tight scoped I think. That might make sense if we were catching all data in memory line you had it, but that comes with it's own problems. So instead we always get fresh data from this service and cache within more dedicated scopes, like pick-ups

Reply:
What problems does it come with? I think it makes more sense to build our classes around the data, rather than the specifics of hows its stored, that way we can switch it out easier if we ever want to, plus it isolates levels of concern.

Reply:
Catching data at this level doesn't make much sense to me. Best to cache at the smallest scope or we end up with too much cache update calls.

Reply:
That makes sense. I can move the caching layer down to the spread sheet level that they all share, that way each sheet is cached once invisibly in the back ground